### ISSUE TITLE
Support simple aggregations on list columns

### ISSUE THREAD

**Issue by DavideCanton** [pola-rs/polars#9859]
### Problem description

Sorry if this is a duplicate, but I did not find anything related by searching in the issue tracker.

Is there any plan regarding supporting simple aggregations (like min or max) on list columns?

Something like a lexicographic order, such that:
- simple list are ordered by comparing value by value, stopping at the shortest one (for example `[1,2] < [1,3]`, `[1,2] < [1,2,3]` and so on).
- nested list are ordered recursively (for example `[[1,2],[1,3]] < [[1,3]]`, `[[1,2]] < [[1,2],[1,3]]` and so on`.

AFAIK, polars by now returns always null in these cases.

**Comment by stinodego** [comment]
This is something we'd like to support, but there are some quirks to work out with regards to (nested) null values.

This is not high priority for us right now.

**Comment by DavideCanton** [comment]
I see, at least now those operation error when executed on list columns, it's better than returning `null` 👍🏻

**Comment by coastalwhite** [comment]
Note that `list.agg` now also exists which will implement basically any aggregation.

**Comment by DavideCanton** [comment]
> Note that `list.agg` now also exists which will implement basically any aggregation.

True, but this does not fit my use case. I don't want to run an aggregation on every element of the list, I want to aggregate a list column

**Comment by cmdlineluser** [comment]
Just adding a code example to avoid any confusion:

```python
import polars as pl

df = pl.DataFrame({"x": [[1,3], [1,2]], "y": [[[1,2],[1,3]], [[1,3]]]})

df.select(pl.all().min())
# InvalidOperationError: `min` operation not supported for dtype `list[i64]`
```

The frame-level methods also still seem to return null:
```
df.min()
# shape: (1, 2)
# ┌───────────┬─────────────────┐
# │ x         ┆ y               │
# │ ---       ┆ ---             │
# │ list[i64] ┆ list[list[i64]] │
# ╞═══════════╪═════════════════╡
# │ null      ┆ null            │
# └───────────┴─────────────────┘
```

Nested types are supported in the newer (min|max)_by methods: https://github.com/pola-rs/polars/pull/26849

```python
df.select(pl.all().min_by(pl.all()))
# shape: (1, 2)
# ┌───────────┬──────────────────┐
# │ x         ┆ y                │
# │ ---       ┆ ---              │
# │ list[i64] ┆ list[list[i64]]  │
# ╞═══════════╪══════════════════╡
# │ [1, 2]    ┆ [[1, 2], [1, 3]] │
# └───────────┴──────────────────┘
```

So it seems like it's just a matter of dispatching to row encoding for regular min/max also?
